packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -0,0 +1,294 @@
+  AvatarResponse,
+  BeginSigninHandler,
+  BeginSigninResponse,
+  BeginSigninResultError,
+  CachedSigninHandler,
+  LocationState,
+  SigninContainerIntegration,
+} from './interfaces';
+import { getCredentials } from 'fxa-auth-client/browser';
+import { GraphQLError } from 'graphql';
+import {
+  AuthUiErrorNos,
+  AuthUiErrors,
+  composeAuthUiErrorTranslationId,
+} from '../../lib/auth-errors/auth-errors';
+import VerificationMethods from '../../constants/verification-methods';
+import VerificationReasons from '../../constants/verification-reasons';
+import AuthenticationMethods from '../../constants/authentication-methods';
+
+/*
+ * In content-server, the `email` param is optional. If it's provided, we
+ * check against it to see if the account exists and if it doesn't, we redirect
+ * users to `/signup`.
+ *
+ * In the React version, we're temporarily always passing the `email` param over
+ * from the Backbone index page until the index page is converted over, in which case
+ * we can pass the param with router state. Since we already perform this account exists
+ * (account status) check on the Backbone index page, which is rate limited since it doesn't
+ * require a session token, we also temporarily pass email status params to 1) signal not to
+ * perform the check again but also because 2) these params are needed to conditionally
+ * display UI in signin. If no status params are passed and `email` is, or we read the
+ * email from local storage, we perform the check and redirect existing user emails to
+ * `/signup` to match content-server functionality.
+ */
+
+const SigninContainer = ({
+  integration,
+  serviceName,
+}: {
+  integration: SigninContainerIntegration;
+  serviceName: MozServices;
+} & RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const navigate = useNavigate();
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state?: LocationState;
+  };
+  const { queryParamModel, validationError } =
+    useValidatedQueryParams(SigninQueryParams);
+
+  // email with either come from React signup (router state),
+  // Backbone index (query param), or will be cached (local storage)
+  const {
+    email: emailFromLocationState,
+    hasLinkedAccount: hasLinkedAccountFromLocationState,
+    hasPassword: hasPasswordFromLocationState,
+  } = location.state || {};
+
+  const [accountStatus, setAccountStatus] = useState({
+    hasLinkedAccount:
+      queryParamModel.hasLinkedAccount || hasLinkedAccountFromLocationState,
+    hasPassword: queryParamModel.hasPassword || hasPasswordFromLocationState,
+  });
+  const { hasLinkedAccount, hasPassword } = accountStatus;
+
+  const nonCachedEmail = queryParamModel.email || emailFromLocationState;
+  let email = nonCachedEmail;
+  let sessionToken: hexstring | undefined;
+  let uid: hexstring | undefined;
+  // only read from local storage if email isn't provided via query param or router state
+  if (!nonCachedEmail) {
```

Comment:
I think we might not be hitting this and get redirected to index as soon as there is no "nonCachedEmail"

Reply:
What are your steps to reproduce?

Reply:
Nevermind me - I ended up in an odd state from creating an account, restarting my stack without signing out, and then going directly to `localhost:3030/signin` and `localhost:3030/signin?showReactApp=true`. In this situation, backbone suggests the previously signed in email (for an account that no longer exists), but react attempts to redirect to `/`, which redirects to backbone signin... Very much an edge case that we probably don't need to account for